### PR TITLE
PLAT-5931: Support node group taints, labels, and tags

### PR DESCRIPTION
**Pull request by fraenkel**
PLAT-5931: Support node group taints, labels, and tags
    PLAT-5931: Support spot instances
    PLAT-5931: Empty the default security group

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -78,7 +74,7 @@ module "storage" {
   efs_access_point_path        = var.efs_access_point_path
   s3_force_destroy_on_deletion = var.s3_force_destroy_on_deletion
   vpc_id                       = local.vpc_id
-  subnets                      = local.private_subnets
+  subnets                      = [for s in local.private_subnets : s.subnet_id]
```
Comment (miguelhar) [inline main.tf:77]: this is a bit confusing as we are defining the subnets in 3 diff places, consider renaming to `private_subnets_ids`

**Review by miguelhar — APPROVED** [review]

File: submodules/eks/variables.tf [hunk 2]
```diff
@@ -68,13 +74,16 @@ variable "default_node_groups" {
...
           instance_type  = optional(string, "g4dn.xlarge")
+          spot           = optional(bool, false)
           min_per_az     = optional(number, 0)
           max_per_az     = optional(number, 10)
           desired_per_az = optional(number, 0)
           labels = optional(map(string), {
             "dominodatalab.com/node-pool" = "default-gpu"
             "nvidia.com/gpu"              = true
           })
+          taints = optional(list(object({ key = string, value = optional(string), effect = string })), [])
```
Comment (steved) [inline submodules/eks/variables.tf:85]: Does this need a default `nvidia.com/gpu` taint as well?

**Review by steved — APPROVED** [review]

File: submodules/eks/main.tf [hunk 3]
```diff
@@ -3,6 +3,7 @@ data "aws_caller_identity" "aws_account" {}
 
 locals {
   eks_cluster_name  = var.deploy_id
+  vpc_id            = var.vpc_id
```
Comment (miguelhar) [inline submodules/eks/main.tf:6]: if these are the same, why not just use `var.vpc_id`

File: submodules/network/outputs.tf [hunk 4]
```diff
@@ -4,11 +4,11 @@ output "vpc_id" {
...
-  value       = [for sb in aws_subnet.private : sb.id]
+  description = "List of private subnet ID and AZ"
+  value       = [for cidr, c in local.private_cidrs : { subnet_id = aws_subnet.private[cidr].id, az = c.az }]
 }
 
 output "public_subnets" {
-  description = "List of IDs of public subnets"
-  value       = [for sb in aws_subnet.public : sb.id]
+  description = "List of public subnet ID and AZ"
+  value       = [for cidr, c in local.public_cidrs : { subnet_id = aws_subnet.public[cidr].id, az = c.az }]
```
Comment (miguelhar) [inline submodules/network/outputs.tf:13]: this will break ocp.

Comment (fraenkel) [reply submodules/network/outputs.tf:13]: in what way?

File: submodules/eks/main.tf (hunk 3 above)
Comment (fraenkel) [reply submodules/eks/main.tf]: done

File: submodules/eks/variables.tf (hunk 2 above)
Comment (fraenkel) [reply submodules/eks/variables.tf:85]: yes, added.

File: submodules/network/outputs.tf (hunk 4 above)
Comment (miguelhar) [reply submodules/network/outputs.tf:13]: https://github.com/dominodatalab/terraform-aws-ocp/blob/main/main.tf#L89
list(string) --> list(object{subnet_id= string,az = string})

File: main.tf (hunk 1 above)
Comment (fraenkel) [reply main.tf:77]: its just subnet_ids.... we dont care if its public or private.

**Pull request merged**